### PR TITLE
Add return value checks for connect

### PR DESCRIPTION
**Pull request by teetak01**
Also moved memcpy to protect it from possible NULL pointer

@yogpan01 @anttiylitokola 

Added some check for connect.

getaddrinfo is also not very well checked so that might need some checking. getaddrinfo also seems to hang quite long if the given address does not exist, possibly could be replaced with getaddrinfo_a which allows specifying timeout.

### CODE REVIEW

File: source/m2mconnectionhandlerpimpl.cpp [hunk 1]
```diff
@@ -93,6 +95,8 @@ bool M2MConnectionHandlerPimpl::resolve_server_address(const String& server_addr
     int r;
     struct sockaddr_in *a = NULL;
     struct sockaddr_in6 *a6 = NULL;
+    int8_t err_connect;
```
Comment (anttiylitokola) [inline source/m2mconnectionhandlerpimpl.cpp]: Initial value missing. This could be set to zero?

**Comment by teetak01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by teetak01** [comment]
@anttiylitokola Closing this.

**Pull request closed without merging**